controller/sql-query-controller.js

```diff
@@ -49,9 +49,31 @@ exports.getAllArticles = async function (req, res, next) {
     catch (err) { next(err) }
 }
 
-exports.getAllComments = async function (req, res, next)
-{
+exports.getAllComments = async function (req, res, next) {
     const allComments = await model.queryAllComments(req.params.article_id);
     try { res.status(200).send({ "comments": allComments }) }
     catch (err) { next(err) }
+}
+
+exports.postNewComment = async function (req, res, next) {
+    const allArticles = await model.queryAllArticles();
+    const allUsers = await model.queryAllUsers();
+
+    //If the article_id is present in the articles table and the username is present in users table
+    // AND if properties username and body exist in the request body
+    if (((allArticles.filter(eachArticle => eachArticle.article_id == req.params.article_id)).length !== 0)
+        &&
+        ((allUsers.filter(eachUser => eachUser.username === req.body.username)).length !== 0)
+        &&
+        req.body.body)
```

Comment:
Here you query all articles and then proceed to do a filter to check if the articles exists.
However in a previous ticket you wrote a function to get an article by article ID which you could of used?

Anything in which you want to handle data should be done in the model and then passed onto the controller